README.md

```diff
@@ -2,157 +2,317 @@
+
+## Get started [↑](#structure)
+
+### Prerequisites [↑](#structure)
+
+To start using Summoner make sure that you have the following tools installed on your machine:
+
+* [`git`](https://git-scm.com) – to initialize the GitHub repo.
+* [`hub`](https://github.com/github/hub) – to upload the project to GitHub.
+* [`curl`](https://curl.haxx.se) – to download licenses.
+
+### Installation [↑](#structure)
+
+Summon contains two executables:
+
+* `summon`: scaffold projects using interactive CLI mode.
+* `summon-tui`: scaffold projects using TUI.
+
+Below you can find the description of how to install each of them so you can choose the one you want.
+
+#### Summon-TUI [↑](#structure)
+
+> **NOTE:** `summon-tui` is not supported on Windows. See [this issue](https://github.com/jtdaugherty/vty/pull/1).
+
+##### TUI: from source [↑](#structure)
+
+Fetch the repo using `cabal`:
+
+```shell
+cabal get -s summoner
+```
+
+or clone the directory from GitHub:
+
+```shell
+git clone https://github.com/kowainik/summoner.git
 ```
-$ source <(summon --bash-completion-script `which summon`)
+
+Step into the directory:
+
+```haskell
+cd summoner
+```
+
+Build and install using `cabal`:
+
+```shell
+cabal new-install summoner-tui:exe:summon-tui
+```
+
+Build and install using `stack`:
+
+```shell
+stack install summoner-tui:exe:summon-tui
 ```
 
-After that, you can call `summon` with the required command. To create a
-project, use `summon new` command specifying the preferred CLI options, follow
-the instructions during the interactive process of the project creation, and a
-new project would be created in a subfolder as well as a repository under your
-GitHub account (if requested).
+#### Summon-CLI [↑](#structure)
 
-### Usage
+##### CLI: from Hackage/Stackage [↑](#structure)
 
-There are several options how to set particular configurations for the new projects:
+Using `cabal`:
+
+```shell
+cabal new-update
+cabal new-install summoner
+```
+
+Using `stack`:
+
+```shell
+stack install summoner
+```
+
+##### CLI: from source [↑](#structure)
+
+Fetch the repo using `cabal`:
+
+```shell
+cabal get -s summoner
+```
+
+or clone the directory from GitHub:
+
+```shell
+git clone https://github.com/kowainik/summoner.git
+```
+
+Step into the directory:
+
+```haskell
+cd summoner
+```
+
+Build and install using `cabal`:
+
+```shell
+cabal new-install summoner-cli:exe:summon
+```
+
+Build and install using `stack`:
+
+```shell
+stack install summoner:exe:summon
+```
+
+### Setup [↑](#structure)
+
+You can turn on the bash auto-completion by running the following command:
+
+```shell
+source <(summon --bash-completion-script `which summon`)
+```
+
+## Usage [↑](#structure)
+
+To start `summon`ing projects nothing additional is required. However, for the more customizable setting, you can use additional configuration settings.
+There are several options how to set the particular configurations for the new projects (in priority increasing order):
```

Comment:
There are several options you can use to set ~~the~~ particular configurations for ~~the~~ new projects (in increasing order of priority)